api/src/main/java/io/strimzi/api/kafka/model/balancing/BrokerCapacity.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.balancing;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import io.strimzi.api.kafka.model.UnknownPropertyPreserving;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.strimzi.crdgenerator.annotations.Maximum;
+import io.strimzi.crdgenerator.annotations.Minimum;
+import io.strimzi.crdgenerator.annotations.Pattern;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Representation of the Cruise Control broker capacity settings. Since the Kafka brokers
+ * in Strimzi are homogeneous, the capacity values for each resource will be
+ * used for every broker.
+ */
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({"disk", "cpuUtilization", "inboundNetwork", "outboundNetwork"})
+@EqualsAndHashCode
+public class BrokerCapacity implements UnknownPropertyPreserving, Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    private String disk;
+    private Integer cpuUtilization;
+    private String inboundNetwork;
+    private String outboundNetwork;
+    private Map<String, Object> additionalProperties = new HashMap<>(0);
+
+    @JsonInclude(JsonInclude.Include.NON_DEFAULT)
+    @Pattern("^[0-9]+([KMGTPE]i?)?$")
```

Comment:
I think we should allow stuff like 1.5T and exponential notation like 2e6 (which Kube does for memory, for example). 

```suggestion
    @Pattern("^[0-9]+([.][0-9]*)?([KMGTPE]i?|e[0-9]+)?$")
```

And I guess to be consistent with the network settings we should allow the B suffix in there too. It's difficult because we're trying to be consistent with Kube, but what Kube does (omitting the actual unit) doesn't really work for the network rates.

Reply:
Updated to include `.` and exponential notation.

I think we should leave the B suffix out of the `disk` capacity values so that it is consistent with the storage values we use for Strimzi (which follow the K8s memory format). I do still think the B suffix is very important for the network rates though! What do you think?